### PR TITLE
Report React context exports

### LINKED ISSUE #53: Report React contexts
As a convention, React contexts usually start with a capital letter, e.g. `MyContext`. This means the plugin considers them a component and not report them, which is not ideal, as they still break HMR.

Example:
```typescript
export const hello = "world";  // reported correctly
export const Component = () => <div>My component</div>;
```
```typescript
export const MyContext = createContext("hello");  // not reported
export const Component = () => <div>Component</div>;
```

It would be really good if the plugin could report these cases (probably with a new config option). I can think of two potential ways to detect React contexts:
* Naming convention only - if the name starts with an capital letter and ends with `Context`
* A constant starting with a capital letter that calls `createContext`

Either one or a combination of both could be used.

Let me know what you think - if you agree this would be a good addition, I could also create a PR.

Thanks!

### PR DESCRIPTION
**Pull request by IgorAufricht**
Fixes #53

### CODE REVIEW

File: src/only-export-components.ts [hunk 1]
```diff
@@ -56,7 +59,7 @@ export const onlyExportComponents: TSESLint.RuleModule<
       checkJS = false,
       allowExportNames,
     } = context.options[0] ?? {};
-    const filename = context.getFilename();
+    const filename = context.filename;
```
Comment (IgorAufricht) [inline src/only-export-components.ts:62]: `context.getFilename()` is deprecated, see https://eslint.org/docs/latest/extend/custom-rules#the-context-object

File: src/only-export-components.test.ts [hunk 2]
```diff
@@ -177,6 +177,18 @@ const valid = [
...
+  {
+    name: "Two components, one of them with 'Context' in its name",
+    code: "export const MyComponent = () => {}; export const ChatContext = () => {};",
+  },
+  {
+    name: "Component & local React context",
+    code: "export const MyComponent = () => {}; const MyContext = createContext('test');",
+  },
+  {
+    name: "Only React context (",
```
Comment (ArnaudBarre) [inline src/only-export-components.test.ts:189]:
```suggestion
    name: "Only React context",
```

File: src/only-export-components.ts [hunk 3]
```diff
@@ -35,6 +36,8 @@ export const onlyExportComponents: TSESLint.RuleModule<
         "Fast refresh only works when a file only exports components. Move your component(s) to a separate file.",
       noExport:
         "Fast refresh only works when a file has exports. Move your component(s) to a separate file.",
+      reactContext:
```
Comment (ArnaudBarre) [inline src/only-export-components.ts:39]: Nice to make a specific message for that

**Review by ArnaudBarre — APPROVED** [review]
Looks great fantastic!

**Pull request merged**